Driver unable to connect to CockroachDB

Looks like something isn't compatible, either cockroachdb or this library.

Following this tutorial:
https://www.cockroachlabs.com/docs/v19.1/build-a-java-app-with-cockroachdb.html#insecure

DataSource snip (`PGDataSource`):
```
    pg.setServerName("localhost");
    pg.setPortNumber(26257);
    pg.setDatabaseName("bank");
    pg.setUser("maxroach");
    pg.setPassword(null);
    pg.setApplicationName("Testing");
```

Stack trace:
```
org.jdbi.v3.core.ConnectionException: com.impossibl.postgres.jdbc.PGSQLSimpleException: Connection Error: java.lang.NullPointerException
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:316)
	at org.jdbi.v3.core.Jdbi.withHandle(Jdbi.java:337)
        at io.grpc.stub.ServerCalls$UnaryServerCallHandler$UnaryServerCallListener.onHalfClose(ServerCalls.java:171)
	at io.grpc.internal.ServerCallImpl$ServerStreamListenerImpl.halfClosed(ServerCallImpl.java:331)
	at io.grpc.internal.ServerImpl$JumpToApplicationThreadServerStreamListener$1HalfClosed.runInContext(ServerImpl.java:808)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.impossibl.postgres.jdbc.PGSQLSimpleException: Connection Error: java.lang.NullPointerException
	at com.impossibl.postgres.jdbc.ErrorUtils.makeSQLException(ErrorUtils.java:137)
	at com.impossibl.postgres.jdbc.ConnectionUtil.createConnection(ConnectionUtil.java:212)
	at com.impossibl.postgres.jdbc.AbstractDataSource.createConnection(AbstractDataSource.java:137)
	at com.impossibl.postgres.jdbc.PGDataSource.getConnection(PGDataSource.java:71)
	at com.impossibl.postgres.jdbc.PGDataSource.getConnection(PGDataSource.java:63)
	at org.jdbi.v3.core.Jdbi.open(Jdbi.java:301)
	... 13 more
Caused by: java.io.IOException: java.lang.NullPointerException
	at com.impossibl.postgres.protocol.v30.ServerConnectionFactory.translateConnectionException(ServerConnectionFactory.java:585)
	at com.impossibl.postgres.protocol.v30.ServerConnectionFactory.connect(ServerConnectionFactory.java:258)
	at com.impossibl.postgres.protocol.v30.ServerConnectionFactory.connect(ServerConnectionFactory.java:150)
	at com.impossibl.postgres.protocol.v30.ServerConnectionFactory.connect(ServerConnectionFactory.java:131)
	at com.impossibl.postgres.system.BasicContext.<init>(BasicContext.java:187)
	at com.impossibl.postgres.jdbc.PGDirectConnection.<init>(PGDirectConnection.java:221)
	at com.impossibl.postgres.jdbc.ConnectionUtil.createConnection(ConnectionUtil.java:203)
	... 17 more
Caused by: java.lang.NullPointerException
	at com.impossibl.postgres.protocol.v30.ServerConnectionFactory$3.handleComplete(ServerConnectionFactory.java:489)
	at com.impossibl.postgres.protocol.v30.StartupRequest$Handler.readyForQuery(StartupRequest.java:195)
	at com.impossibl.postgres.protocol.v30.MessageDispatchHandler.receiveReadyForQuery(MessageDispatchHandler.java:621)
	at com.impossibl.postgres.protocol.v30.MessageDispatchHandler.parseAndDispatch(MessageDispatchHandler.java:359)
	at com.impossibl.postgres.protocol.v30.MessageDispatchHandler.dispatch(MessageDispatchHandler.java:192)
	at com.impossibl.postgres.protocol.v30.MessageDispatchHandler.channelRead(MessageDispatchHandler.java:147)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:321)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:295)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:352)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1408)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:374)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:360)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:930)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	... 1 more
```

--- a/driver/src/main/java/com/impossibl/postgres/jdbc/CancelRequestTask.java
+++ b/driver/src/main/java/com/impossibl/postgres/jdbc/CancelRequestTask.java
@@ -44,10 +44,13 @@ import java.net.InetSocketAddress;
 import java.net.Socket;
 import java.net.SocketAddress;
 import java.nio.ByteBuffer;
+import java.util.logging.Logger;
 
 import io.netty.channel.unix.DomainSocketAddress;
 
 public class CancelRequestTask extends ExecutionTimerTask {
+
+  private static final Logger logger = Logger.getLogger(CancelRequestTask.class.getName());
 
   private SocketAddress serverAddress;
   private ServerConnection.KeyData keyData;
@@ -63,6 +66,11 @@ public class CancelRequestTask extends ExecutionTimerTask {
   }
 
   private void sendCancelRequest() {
+
+    if (keyData.getProcessId() == 0 && keyData.getSecretKey() == 0) {
+      logger.warning("Cannot send CancelRequest because of missing BackendKeyData.");
+      return;
+    }
 
     try {
 

--- a/driver/src/main/java/com/impossibl/postgres/protocol/v30/ServerConnectionFactory.java
+++ b/driver/src/main/java/com/impossibl/postgres/protocol/v30/ServerConnectionFactory.java
@@ -495,7 +495,7 @@ public class ServerConnectionFactory implements com.impossibl.postgres.protocol.
       }
 
       @Override
-      public void handleComplete(Integer processId, Integer secretKey, Map<String, String> parameterStatuses, List<Notice> notices) {
+      public void handleComplete(int processId, int secretKey, Map<String, String> parameterStatuses, List<Notice> notices) {
 
         startupParameterStatuses.putAll(parameterStatuses);
         startupKeyData.set(new KeyData(processId, secretKey));

--- a/driver/src/main/java/com/impossibl/postgres/protocol/v30/StartupRequest.java
+++ b/driver/src/main/java/com/impossibl/postgres/protocol/v30/StartupRequest.java
@@ -63,7 +63,7 @@ public class StartupRequest implements ServerRequest {
 
     void handleNegotiate(Version maxProtocolVersion, List<String> unrecognizedParameters) throws IOException;
 
-    void handleComplete(Integer processId, Integer secretKey, Map<String, String> parameterStatuses, List<Notice> notices) throws IOException;
+    void handleComplete(int processId, int secretKey, Map<String, String> parameterStatuses, List<Notice> notices) throws IOException;
     void handleError(Throwable cause, List<Notice> notices) throws IOException;
 
   }
@@ -71,8 +71,8 @@ public class StartupRequest implements ServerRequest {
   private Version protocolVersion;
   private Map<String, Object> startupParameters;
   private CompletionHandler handler;
-  private Integer backendProcessId;
-  private Integer backendSecretKey;
+  private int backendProcessId;
+  private int backendSecretKey;
   private Map<String, String> parameterStatuses;
   private List<Notice> notices;
 

